### PR TITLE
1848 cross validation make sure all award references in federal awards are unique

### PR DESCRIPTION
**Pull request by sambodeme**
## Description
#1848 

To test locally, run:
`make docker-clean`
`make docker-first-run`
`docker compose up`
make sure backend-web is running properly and run this cmd in another terminal run
`docker compose run web python manage.py test audit.cross_validation.test_check_award_reference_uniqueness`

## PR checklist: submitters

- [x]   Link to an issue if possible. If there’s no issue, describe what your branch does. Even if there is an issue, a brief description in the PR is still useful.
- [ ]   List any special steps reviewers have to follow to test the PR. For example, adding a local environment variable, creating a local test file, etc.
- [ ]   For extra credit, submit a screen recording like [this one](https://github.com/GSA-TTS/FAC/pull/1821).
- [x]   Make sure you’ve merged `main` into your branch shortly before creating the PR. (You should also be merging `main` into your branch regularly during development.)
- [x]   Make sure that whatever feature you’re adding has tests that cover the feature. This includes test coverage to make sure that the previous workflow still works, if applicable.
- [ ]   Do manual testing locally. Our tests are not good enough yet to allow us to skip this step.
- [ ]   If Git surgery is necessary at any point, repeat the testing after it’s finished.
- [ ]   Once a PR is merged, keep an eye on it until it’s deployed to dev, and do enough testing on dev to verify that it deployed successfully, the feature works as expected, and the happy path for the broad feature area (such as submission) still works.

## PR checklist: reviewers

- [x]   Pull the branch to your local environment and run `make docker clean; make docker-first-run && docker compose up`; then run `docker compose exec web /bin/bash -c "python manage.py test"`
- [ ]   Manually test out the changes locally.
- [x]   Check that the PR has appropriate tests. Look out for changes in HTML/JS/JSON Schema logic that may need to be captured in Python tests even though the logic isn’t in Python.
- [ ]   If Git surgery is necessary at any point (such as during a merge party), repeat the testing after it’s finished.
        
The larger the PR, the stricter we should be about these points.

### CODE REVIEW

**Review by danswick — APPROVED** [review]
Looks good! 

Tests all pass locally. I didn't do a complete manual test, mostly because I don't yet have a complete set of test workbooks to upload / get to the cross-validation step.

File: backend/audit/cross_validation/check_award_reference_uniqueness.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    """
+
+    all_sections = sac_dict.get("sf_sac_sections", {})
+    federal_awards_section = all_sections.get("federal_awards") or {}
+    federal_awards = federal_awards_section.get("federal_awards", [])
+    award_refs = []
+    errors = []
+
+    for award in federal_awards:
+        award_reference = award.get("award_reference", None)
```
Comment (tadhg-ohiggins) [inline backend/audit/cross_validation/check_award_reference_uniqueness.py:19]: You don't need the `None` argument here; `.get` will return `None` as the default by default.

Comment (jadudm) [inline backend/audit/cross_validation/check_award_reference_uniqueness.py:19]:
```suggestion
        award_reference = award.get("award_reference")

```

Comment (sambodeme) [reply backend/audit/cross_validation/check_award_reference_uniqueness.py]: Agreed. I specifically added this to remind us to discuss and create a follow-up task regarding award_ref being mandatory. What we did and already implemented so you 100% right here, None is not necessary.

**Pull request merged**